pkg/ibmvpc/vpc.go

```diff
@@ -346,6 +368,23 @@ func (n *NACL) AllowedConnectivity(src, dst vpcmodel.Node, isIngress bool) (*com
 	return n.analyzer.AllowedConnectivity(subnetCidr, inSubnetCidr, targetNode.Cidr(), isIngress)
 }
 
+func (n *NACL) RulesInConnectivity(src, dst vpcmodel.Node, conn *common.ConnectionSet, isIngress bool) ([]int, error) {
+	targetNode, subnetCidr, inSubnetCidr, err := n.initConnectivityComputation(src, dst, isIngress)
+	if err != nil {
+		return nil, err
+	}
+	// check if the subnet of the given node is affected by this nacl
+	if _, ok := n.subnets[subnetCidr]; !ok {
+		return nil, nil // not affected by current nacl
+	}
+	// nacl has no control on traffic between two instances in its subnet; this is marked by a rule with index -1
+	// which is not printed but only signals that this filter does not block (since there are rules)
+	if allInSubnet, err := common.IsAddressInSubnet(targetNode.Cidr(), subnetCidr); err == nil && allInSubnet {
+		return []int{-1}, nil
+	}
```

Comment:
@adisos do you remember why this error is ignored? 
Because when its not ignored in `func (n *NACL) AllowedConnectivity(src, dst vpcmodel.Node, isIngress bool) `
all (non-explainability) tests fail

Reply:
I assume this requires more investigation, maybe open a separate issue for this? 
If the tests fail when it is not ignored, it probably indicates some problem.

Reply:
@ShiriMoran do you check this?

Reply:
I have some guesses why it is happening, but it needs to be checked. Not as part of this PR - the relevant part of the code is not of this PR. The function above, from which the code was taken, is not of this PR.
https://github.com/np-guard/vpc-network-config-analyzer/issues/344
I assigned myself.